pages/search-result-tokens/index.tsx

```diff
@@ -0,0 +1,225 @@
+      }),
+    { refetchInterval: 10000 },
+  )
+
+  const handleTokenLink = (type: GraphQLSchema.TokenType, tokenId: string, contractAddr?: string) => {
+    switch (type) {
+      case GraphQLSchema.TokenType.ERC20:
+        return `/token/${tokenId}`
+      case GraphQLSchema.TokenType.ERC721:
+        return `/nft-item/${contractAddr}/${tokenId}`
+      case GraphQLSchema.TokenType.ERC1155:
+        return `/multi-token-item/${contractAddr}/${tokenId}`
+      default:
+        return ''
+    }
+  }
+
+  return (
+    <>
+      <SubpageHead subtitle={title} />
+      <div className={styles.container}>
+        <PageTitle>{title}</PageTitle>
+        <div className={styles.list}>
+          <div className={styles.subheader}>
+            <span>
+              {t(`n_kinds_in_total`, {
+                ns: 'list',
+                number: list?.metadata.total_count.toLocaleString('en') ?? '-',
+              })}
+            </span>
+            {list?.metadata.total_count ? <Pagination {...list.metadata} /> : null}
+          </div>
+          <Table>
+            <thead>
+              <tr>
+                <th className={styles.tokenHeader}>{t('token', { ns: 'list' })}</th>
+                <th className={styles.typeHeader}>{t('type', { ns: 'list' })}</th>
+                <th>{t('address')} </th>
+              </tr>
+            </thead>
+            <tbody>
+              {list?.metadata.total_count ? (
+                list.entries.map(item => {
+                  return (
+                    <tr key={item.id}>
+                      {handleTokenLink(item.eth_type, item.id, item.contract_address_hash) ? (
+                        <td title={item.name}>
+                          <NextLink href={handleTokenLink(item.eth_type, item.id, item.contract_address_hash)}>
+                            <a className={styles.token}>
+                              <TokenLogo name={item.name} logo={item.icon} />
+                              <span>
+                                {item.name ?? '-'}
+                                {item.symbol ? `(${item.symbol})` : ''}
+                              </span>
+                            </a>
+                          </NextLink>
+                        </td>
+                      ) : (
+                        <td title={item.name}>
+                          <span className={styles.token}>
+                            <TokenLogo name={item.name} logo={item.icon} />
+                            <span>
+                              {item.name ?? '-'}
+                              {item.symbol ? `(${item.symbol})` : ''}
+                            </span>
+                          </span>
+                        </td>
+                      )}
+                      <td className={styles.type}>{t(item.eth_type as string, { ns: 'account' })}</td>
+                      {item?.contract_address_hash ? (
+                        <td className={styles.addr} title={item.contract_address_hash}>
```

Comment:
It feels better to deconstruct the `item.contract_address_hash` because the same expression is used so many times.